StyleCop.Analyzers/StyleCop.Analyzers/LayoutRules/SA1514ElementDocumentationHeaderMustBePrecededByBlankLine.cs

```diff
@@ -170,6 +170,18 @@ private static void HandleDeclaration(SyntaxNodeAnalysisContext context)
                     // no leading blank line necessary at start of scope.
                     return;
                 }
+
+                // Logic to handle global namespace case
+                if (prevToken.IsKind(SyntaxKind.None))
+                {
+                    // Check if the node is the first non-trivia element in the file
+                    var firstToken = context.Node.SyntaxTree.GetRoot().GetFirstToken();
+                    if (firstToken == context.Node.GetFirstToken())
+                    {
+                        // Node is the first element in the global namespace
+                        return;
```

Comment:
I suspect that you skip too many cases as the code looks right now. I suggest you add tests for cases where it **should** trigger even though there is no token before the documentation. I am thinking e.g. a normal comment or a directive before the documentation. And if that reveals a need for a more subtle bail-out, consider similar cases where it should *not* trigger.

Reply:
Yep. Great catch. I need to review and add some more tests. Thanks